frontend-build-pipeline.yml

```diff
@@ -0,0 +1,52 @@
+trigger:
+- main
+- staging
+- development
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+
+# Node.js is required to build Angular apps
+- task: NodeTool@0
+  inputs:
+    versionSpec: '12.14'
+  displayName: 'Install Node.js'
+
+# --dev option installs devDependencies from package.json
+# probably needed for running the Karma tests
+#
+# angular-devkit/build-angular
+# https://stackoverflow.com/questions/50333003/could-not-find-module-angular-devkit-build-angular
+- script: |
+    npm install --save-dev @angular-devkit/build-angular
+    npm install -g @angular/cli
+    npm install
+    ng build --prod
+  displayName: 'npm Install and Build'
```

Comment:
cf [my review comments here](https://github.com/1058-Sophia-Gavrila-Forge/Portfolio-Frontend/pull/3#pullrequestreview-708776485). also:
- `@angular-devkit/build-angular` is already there in the package.json so re-adding it should be superfluous.
- installing the angular cli globally only to install it locally as well is a huge waste of time only to get to run `ng` instead of `npx ng` (i don't expect you to have any context for that)

so, this should be replaced by an `npm ci` step and an `npx ng build --prod` step.

Reply:
`@angular-devkit/build-angular` was there because when I was testing with a windows build agent the pipeline would fail without it. I have removed it now since we switched back to using Ubuntu.

Thanks for the information about `ng` and `npx ng`. I'll look into those. Changed `ng` to `npx ng`.

Got rid of `npm install` in favor of `npm ci`.

Reply:
that's strange. that that fixed it is probably a strange side-effect of the actual cause. in a case like this you can mark 'resolve conversation' after your reply.

Reply:
Yeah I was trying out the Windows build agent on a hunch that Chrome vs ChromeHeadless could have been causing the unit tests to fail, but the results were the same. For some reason building on Ubuntu would fail without ChromeHeadless, but it worked fine as Chrome on the Windows build agent.

Sorry I forgot to address this when I resolved the conversation earlier: "cf my review comments here." I don't think I've heard the term cf before, or I just don't recall the acronym right now. What does cf mean?

Reply:
- the windows build agents have everything needed to run GUI apps on. ubuntu build agents do not.
- https://en.wiktionary.org/wiki/cf.#English
- `ng build --prod` used to be the standard, but i just read they deprecated that option in favor of `ng build --configuration production` (or `-c production`, but i suggest long names for readability)

Reply:
update, that changed in angular 12 alongside some other things i didn't notice, with the net effect that `ng build` (or `npm run build`) by itself is fine, no mention of prod needed because it's the new default.

Reply:
Alright thanks again for pointing things out. Changed `ng build --prod` to `ng build`.